apps/etterlatte-oppdater-behandling/src/main/kotlin/InntektsjusteringInfobrevRiver.kt

```diff
@@ -0,0 +1,105 @@
+import rapidsandrivers.InntektsjusteringHendelseType
+import rapidsandrivers.RapidEvents.ANTALL
+import rapidsandrivers.RapidEvents.EKSKLUDERTE_SAKER
+import rapidsandrivers.RapidEvents.KJOERING
+import rapidsandrivers.RapidEvents.SPESIFIKKE_SAKER
+import rapidsandrivers.antall
+import rapidsandrivers.ekskluderteSaker
+import rapidsandrivers.kjoering
+import rapidsandrivers.saker
+
+internal class InntektsjusteringInfobrevRiver(
+    rapidsConnection: RapidsConnection,
+    private val behandlingService: BehandlingService,
+    private val featureToggleService: FeatureToggleService,
+) : ListenerMedLogging() {
+    private val logger = LoggerFactory.getLogger(this::class.java)
+
+    init {
+        initialiserRiver(rapidsConnection, InntektsjusteringHendelseType.SEND_INFOBREV) {
+            validate { it.requireKey(KJOERING) }
+            validate { it.requireKey(ANTALL) }
+            validate { it.interestedIn(SPESIFIKKE_SAKER) }
+            validate { it.interestedIn(EKSKLUDERTE_SAKER) }
+        }
+    }
+
+    override fun haandterPakke(
+        packet: JsonMessage,
+        context: MessageContext,
+    ) {
+        if (!featureToggleService.isEnabled(InntektsjusterinFeatureToggle.START_INFOBREV_INNTEKTSJUSTERING, false)) {
+            logger.info("Utsending av informasjonsbrev er deaktivert ved funksjonsbryter. Avbryter forespørsel.")
+            return
+        }
+
+        val kjoering = packet.kjoering
+        val antall = packet.antall
+
+        val sakType = SakType.OMSTILLINGSSTOENAD
+        // TODO: må vi filtrere på mer? ta bort saker som opphøres innværende år?
+
+        logger.info("$kjoering: Starter ")
+        kjoerIBatch(
+            logger = logger,
+            antall = antall,
+            finnSaker = { antallIDenneRunden ->
+                behandlingService.hentAlleSaker(
+                    kjoering,
+                    antallIDenneRunden,
+                    packet.saker,
+                    packet.ekskluderteSaker,
+                    sakType,
+                )
+            },
+            haandterSaker = { sakerSomSkalInformeres ->
+                sakerSomSkalInformeres.saker.forEach {
+                    journalfoerOgDistribuer(it, kjoering, packet, context)
+                }
+            },
+        )
+
+        logger.info("$kjoering: Ferdig")
+    }
+
+    private fun journalfoerOgDistribuer(
+        sak: Sak,
+        kjoering: String,
+        packet: JsonMessage,
+        context: MessageContext,
+    ) {
+        logger.info("$kjoering: Journalfører og distribuerer infobrev for sak ${sak.id}")
+        behandlingService.lagreKjoering(sak.id, KjoeringStatus.STARTA, kjoering)
+        packet.setEventNameForHendelseType(InntektsjusteringHendelseType.SEND_INFOBREV)
+        packet.sakId = sak.id
+        packet[BREVMAL_RIVER_KEY] = Brevkoder.OMS_INNTEKTSJUSTERING
+        context.publish(packet.toJson())
+        behandlingService.lagreKjoering(sak.id, KjoeringStatus.FERDIGSTILT, kjoering)
```

Comment:
Bør ha en egen river for å ferdigstille som lytter på distribuert brev eller noe sånt? Nå vil den få status ferdigstilt selv om brev skulle feile.